src/core/thread/meshcop_dataset.cpp

```diff
@@ -121,6 +121,32 @@ void Dataset::Get(otOperationalDataset &aDataset)
             break;
         }
 
+        case Tlv::kChannelMask:
+        {
+            uint8_t length = cur->GetLength();
+            const uint8_t *entry = reinterpret_cast<const uint8_t *>(cur) + sizeof(Tlv);
+            const uint8_t *entryEnd =  entry + length;
+
+            while (entry < entryEnd)
+            {
+                switch (reinterpret_cast<const ChannelMaskEntry *>(entry)->GetChannelPage())
```

Comment:
nit: this would be more readable if it was just an if-statement (since there is only one case to handle).